app/web/src/store/actions.store.ts

```diff
@@ -303,43 +303,90 @@ export const useActionsStore = () => {
               eventType: "ChangeSetApplied",
               callback: (_update) => {
                 this.LOAD_ACTION_BATCHES();
+                // Short term fix for reactivity issue on apply, since the
+                // first load won't have the actions since the rebaser isnt done
+                setTimeout(() => this.LOAD_ACTION_BATCHES(), 500);
               },
             },
             {
-              eventType: "ActionAdded",
-              callback: () => {
-                this.FETCH_QUEUED_ACTIONS();
+              eventType: "DeprecatedActionAdded",
+              callback: (action) => {
+                this.rawProposedActionsById[action.id] = action
               },
             },
             {
-              eventType: "ActionRemoved",
-              callback: () => {
-                this.FETCH_QUEUED_ACTIONS();
+              eventType: "DeprecatedActionRemoved",
+              callback: (actionId) => {
+                delete this.rawProposedActionsById[actionId];
               },
             },
             {
               eventType: "DeprecatedActionRunnerReturn",
-              callback: (update) => {
+              callback: async (update) => {
+                const status = update.resource ? (update.resource.status ?? ResourceHealth.Unknown) : ResourceHealth.Error;
                 trackEvent("action_runner_return", {
                   action_runner: update.action,
-                  action_status: update.status,
+                  action_status: status,
                   action_runner_id: update.id,
                   action_batch_id: update.batchId,
                 });
 
-                this.LOAD_ACTION_BATCHES();
+                const batchIndex = this.actionBatches.findIndex((batch) => batch.id === update.batchId);
+                const batch = this.actionBatches[batchIndex];
+                if (!batch) {
+                  // Short term fix for reactivity issue on apply, since the
+                  // first load won't have the actions since the rebaser isnt done
+                  setTimeout(() => this.LOAD_ACTION_BATCHES(), 500);
```

Comment:
This is horrifying but was the only way to consistently get the data after change-set apply. It's a general bug after apply, where data is loaded before the rebaser finishes so everything is stale, but it made actions pretty much unusable, so this is a temporary fix until we solve it for everything.

Documented on https://linear.app/system-initiative/issue/BUG-105/no-reactivity-after-applying-a-change-set

Reply:
I think a 500ms post execution makes sense to me as a patch until we solve it fully!